pkg/api/v1alpha1/cluster_types.go

```diff
@@ -161,14 +161,31 @@ func TaintsSliceEqual(s1, s2 []corev1.Taint) bool {
 	return true
 }
 
+func LabelsMapEqual(s1, s2 map[string]string) bool {
+	if len(s1) != len(s2) {
+		return false
+	}
+	for key, val := range s2 {
+		v, ok := s1[key]
+		if !ok {
+			return false
+		}
+		if val != v {
+			return false
+		}
+	}
+	return true
+}
+
 func (n *ControlPlaneConfiguration) Equal(o *ControlPlaneConfiguration) bool {
 	if n == o {
 		return true
 	}
 	if n == nil || o == nil {
 		return false
 	}
-	return n.Count == o.Count && n.Endpoint.Equal(o.Endpoint) && n.MachineGroupRef.Equal(o.MachineGroupRef) && TaintsSliceEqual(n.Taints, o.Taints)
+	return n.Count == o.Count && n.Endpoint.Equal(o.Endpoint) && n.MachineGroupRef.Equal(o.MachineGroupRef) &&
```

Comment:
we'll want to shorten this chain of validations up at some point, I don't think it really matters right now but it's becoming pretty long